test/Confluent.Kafka.IntegrationTests/Tests/Consumer_MissingCommits.cs

```diff
@@ -0,0 +1,136 @@
+                {
+                    for (int i=0; i<numPartitions; ++i)
+                    {
+                        for (int j=0; j<numMessagesPerPartition; ++j)
+                        {
+                            producer.Produce(new TopicPartition(topic.Name, i), new Message<Null, string> { Value = "test" });
+                        }
+                    }
+                    producer.Flush();
+
+                    for (int i=0; i<numPartitionsWithCommittedOffsets; ++i)
+                    {
+                        consumer.Commit(new TopicPartitionOffset[] { new TopicPartitionOffset(topic.Name, i, commitOffset) });
+                    }
+                }
+
+                var consumers = new List<IConsumer<Null, string>>();
+                var messageCounts = new Dictionary<int, int>();
+                try
+                {
+                    for (int i=0; i<numConsumers; ++i)
+                    {
+                        var consumer = new ConsumerBuilder<Null, string>(new ConsumerConfig { BootstrapServers = bootstrapServers, GroupId = groupId, AutoOffsetReset = AutoOffsetReset.Earliest, Debug = "all" }).Build();
+                        consumers.Add(consumer);
+                        consumer.Subscribe(topic.Name);
+                    }
+
+                    var startTime = DateTime.UtcNow;
+                    var complete = false;
+                    while (!complete)
+                    {
+                        if (DateTime.UtcNow - startTime > TimeSpan.FromSeconds(30))
+                        {
+                            Assert.False(true, "Timed out waiting for consumption of messages to complete");
+                            complete = true;
+                        }
+
+                        for (int i=0; i<numConsumers; ++i)
+                        {
+                            var cr = consumers[i].Consume(0);
+                            if (cr != null && cr.Message != null)
+                            {
+                                if (!messageCounts.ContainsKey(cr.Partition))
+                                {
+                                    messageCounts.Add(cr.Partition, 0);
+                                }
+                                messageCounts[cr.Partition] += 1;
+                            }
+                        }
+
+                        complete = true;
+                        for (int i=0; i<numPartitionsWithCommittedOffsets; ++i)
+                        {
+                            if (!messageCounts.ContainsKey(i) || messageCounts[i] < numMessagesPerPartition-commitOffset)
+                            {
+                                complete = false;
+                                break;
+                            }
+                        }
+                        for (int i=numMessagesPerPartition; i<numPartitions; ++i)
```

Comment:
shouldn't this be `i = numPartitionsWithCommittedOffsets` ?

Reply:
yep, whoops.

at this point i'd also like to note that i was checking messageCounts in the debugger, and could see everything was as expected.